### PR TITLE
feat(permissions): permissions for admin users or with ignored access are now universal

### LINKED ISSUE #7999: ```elgg_override_permissions()``` priority
Any handler registered for 'container_permissions_check' is potentially overlooking access overrides. I have been writing these handlers for years, and it's the first time that I actually realized that I should be including explicit checks for `elgg_is_admin_user()` and `elgg_check_access_overrides()` if I want to preserve the return value (I can't assume that true should be passed along). 
Seems counter-intuitive. I would almost say that we should not even trigger permissions check hooks if access overrides are on.

### PR DESCRIPTION
**Pull request by hypeJunction**
BREAKING CHANGE
User capabilities service will no longer trigger permissions check hooks, when
a) permissions are checked for an admin user
b) permissions are check when access is ignored

Fixes #7999

### CODE REVIEW

File: engine/classes/ElggEntity.php [hunk 1]
```diff
@@ -409,10 +409,12 @@ public function setMetadata($name, $value, $value_type = '', $multiple = false,
 				// need to remove access restrictions right now to delete
 				// because this is the expected behavior
 				$ia = elgg_set_ignore_access(true);
-				if (false === elgg_delete_metadata($options)) {
+				$delete_result = elgg_delete_metadata($options);
```
Comment (hypeJunction) [inline engine/classes/ElggEntity.php]: @jdalsem @jeabakker Could you fix this in 2.3?

File: engine/classes/Elgg/Database/Plugins.php [hunk 2]
```diff
@@ -152,6 +152,7 @@ function generateEntities() {
 	
 		$physical_plugins = $this->getDirsInDir($mod_dir);
 		if (!$physical_plugins) {
+			elgg_set_ignore_access($old_ia);
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/Plugins.php]: @jdalsem @jeabakker Could you fix this in 2.3?

**Comment by hypeJunction** [comment]
@beck24 Can see if it fixes #11062

**Comment by hypeJunction** [comment]
Now the truth with simpletests is coming to surface. Running all tests as admin logged in is just bad.

**Comment by mrclay** [comment]
How would this affect Arck roles, where a user may be "admin" but with many capabilities removed by hook? (as far as I understand it)

**Comment by mrclay** [comment]
From the description it sounds like we're making admin equivalent to access off.

**Comment by hypeJunction** [comment]
We can add a hook to access overrides check rather than stacking permissions.

**Comment by hypeJunction** [comment]
I don't think you can effectively use arck roles for downgrading admin privileges, you would be working your way up: create a regular user with a role that extends an admin role and alters permissions.
But yes, a valid point. In all the years working with Elgg, I never had to mess with admin permissions, but I did have to debug a ton of code to narrow it down to that removed hook and bad stacking of hooks and permissions.
I'd rather we start working towards a proper capabilities service. For the sake of testability and consistency, I need this in. If anyone wants to alter admin permissions, they can make a pull request with a clean solution. What we had in core wasn't clean and invited all kinds of bugs.

**Comment by mrclay** [comment]
> I don't think you can effectively use arck roles for downgrading admin privileges, you would be working your way up: create a regular user with a role that extends an admin role and alters permissions.

So the Roles' "[Admin role](https://github.com/arckinteractive/Roles#implementing-the-moderator-role)" is not actually making users admins then limiting their power? I guess I was confused by that.

**Comment by hypeJunction** [comment]
It could be used both ways. Depends on. How you approach it

**Pull request merged**